sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -6693,25 +6791,32 @@ pi_result piEnqueueEventsWaitWithBarrier(pi_queue Queue,
          It != Queue->CommandListMap.end(); ++It)
```

Comment:
Please double check that we indeed keep immediate command-lists (from all threads) in this `Queue->CommandListMap`

Reply:
Perhaps we have to do the same as in other places: https://github.com/intel/llvm/pull/7041/files#diff-15dd1eb076d2164bd9e87d9057f05f652a716498e8cdf5975e564c65309a0985R6976

Reply:
Yes, all commandlists go into the map.

Reply:
Then maybe use it everywhere instead of iterating with 2 for-loops?

Reply:
A queue has a map by TID of compute queue groups and a map by TID of copy queue groups. So we have to first iterate over the two groups (compute and copy) and then per-thread entries in each map.

Reply:
yes, I understand the reason for 2-deep loopnest, but here we get away by just iterating over `CommandListMap` and assume we see all the same command-lists, right?

Reply:
For standard commandlists there are many commandlists per ZeQueue (L0 queue).
For immediate commandlists there is only one immediate commandlist per L0 queue.
So, for immediate commandlists either type of scan yields the same set of L0 queues.
For standard commandlists if we used CommandListMap we would have to extract the unique L0 queues from the set of commandlists in the map.

You will notice that where ever a two-level for-loop structure is used, the separation between standard/immediate commandlists is inside. In this one case, we check standard/immediate outside, and then do a scan in different ways. This simply matches the code as it was before.
For consistency, this particular instance could also be written as a two-level for-loop with the standard/immediate check inside, if you wish.

Reply:
> For consistency, this particular instance could also be written as a two-level for-loop with the standard/immediate check inside, if you wish.

Yes, that's what I was looking for, consistency. Either using one way or another everywhere (and using a single list is probably a faster option). BTW, why do we need to keep the immediate command-lists in the `CommandListMap` at all?
Perhaps, if we could stop doing that then there would not be this confusing diff.

Having said that, this is NIT and should not block this PR.

Reply:
OK, I've changed it for consistency.
The main reason immediate commandlists are also entered into the map is that throughout the plugin, commandlists are passed around as an iterator on that map, never as a commandlist itself. So, to retain that interface I put immediate commandlists in that map too.

Reply:
That makes sense, thanks!